pkg/controllers/provisioning/binpacking/packable.go

```diff
@@ -39,10 +41,36 @@ type Result struct {
 	unpacked []*v1.Pod
 }
 
+type Packables []*Packable
+
+func (packables Packables) Len() int {
+	return len(packables)
+}
+
+func (packables Packables) Swap(i, j int) {
+	packables[i], packables[j] = packables[j], packables[i]
+}
+
+func (packables Packables) Less(a int, b int) bool {
+	// Check GPU equality assuming GPU classes are mutually exclusive
+	if packables[a].AMDGPUs().Equal(*packables[b].AMDGPUs()) ||
+		packables[a].NvidiaGPUs().Equal(*packables[b].NvidiaGPUs()) ||
+		packables[a].AWSNeurons().Equal(*packables[b].AWSNeurons()) {
+		if packables[a].CPU().Equal(*packables[b].CPU()) {
+			// check for memory
+			return packables[a].Memory().Cmp(*packables[b].Memory()) == -1
+		}
+		return packables[a].CPU().Cmp(*packables[b].CPU()) == -1
+	}
+	return packables[a].AMDGPUs().Cmp(*packables[b].AMDGPUs()) == -1 ||
+		packables[a].NvidiaGPUs().Cmp(*packables[b].NvidiaGPUs()) == -1 ||
+		packables[a].AWSNeurons().Cmp(*packables[b].AWSNeurons()) == -1
```

Comment:
Do the number of AWSNeurons, AMDGPUs, and NvidiaGPUs on each instance increase with instance size at the same rate? If not, wouldn't it be possible for this to have non-deterministic behavior?

Reply:
They increase, not necessarily at the same rate by gpu but by vcpu and memory.  I'm not sure how that would introduce non-deterministic behavior though? 

For example:

```
$ ec2-instance-selector  -o table-wide --allow-list g4ad
Instance Type  VCPUs   Mem (GiB)  Hypervisor  Current Gen  Hibernation Support  CPU Arch  Network Performance  ENIs    GPUs    GPU Mem (GiB)  GPU Info             On-Demand Price/Hr
-------------  -----   ---------  ----------  -----------  -------------------  --------  -------------------  ----    ----    -------------  --------             ------------------
g4ad.xlarge    4       16         nitro       true         false                x86_64    Up to 10 Gigabit     2       1       8              AMD Radeon Pro V520  -No Price Filter Specified-
g4ad.2xlarge   8       32         nitro       true         false                x86_64    Up to 10 Gigabit     2       1       8              AMD Radeon Pro V520  -No Price Filter Specified-
g4ad.4xlarge   16      64         nitro       true         false                x86_64    Up to 10 Gigabit     3       1       8              AMD Radeon Pro V520  -No Price Filter Specified-
g4ad.8xlarge   32      128        nitro       true         false                x86_64    15 Gigabit           4       2       16             AMD Radeon Pro V520  -No Price Filter Specified-
g4ad.16xlarge  64      256        nitro       true         false                x86_64    25 Gigabit           8       4       32             AMD Radeon Pro V520  -No Price Filter Specified-
```